### PR TITLE
Tweak autoinstall behaviour

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

Only enable autoinstall when explicitly enabled when using `packageManager.resolve`, currently when passing undefined and null it also autoinstalls, which might be a problem with third party plugins.

This PR also disabled autoinstall completely when using the build command as this command should probably be reliable across machines (especially build pipelines and CI/CD)

Slightly related to #2978 

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

File: packages/core/parcel/src/cli.js [hunk 1]
```diff
@@ -270,6 +270,7 @@ async function normalizeOptions(command): Promise<InitialParcelOptions> {
   let nodeEnv;
   if (command.name() === 'build') {
     nodeEnv = initialNodeEnv || 'production';
+    command.autoinstall = false;
```
Comment (mischnic) [inline packages/core/parcel/src/cli.js:273]: I would say this should rather be in `@parcel/core`'s `resolveOptions`

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 6.29s | -57.00ms |
| Cached | 2.33s | +80.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 224.00ms | -16.00ms 🚀 |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 224.00ms | -15.00ms 🚀 |
| dist/legacy/index.html | 701.00b | +0.00b | 589.00ms | -66.00ms 🚀 |
| dist/modern/index.html | 701.00b | +0.00b | 587.00ms | -66.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 206.00ms | -35.00ms 🚀 |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 206.00ms | -35.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 28.44s | -163.00ms |
| Cached | 2.47s | +47.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 165.00ms | +12.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 478.42kb | +0.00b | 338.00ms | -20.00ms 🚀 |
| dist/PermalinkedComment.c85d137d.js | 4.22kb | +0.00b | 328.00ms | -28.00ms 🚀 |
| dist/UserProfile.30e15baf.js | 1.62kb | +0.00b | 328.00ms | -28.00ms 🚀 |
| dist/NotFound.5aa6b041.js | 387.00b | +0.00b | 328.00ms | -28.00ms 🚀 |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 260.00ms | -19.00ms 🚀 |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.95m | +4.66s |
| Cached | 5.04s | -315.00ms 🚀 |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/card.ce67dc35.js | 55.09kb | +0.00b | 13.37s | -1.16s 🚀 |
| dist/Modal.e3bd77d0.js | 41.40kb | +0.00b | 6.46s | +998.00ms ⚠️ |
| dist/component.59edff1f.js | 30.78kb | +0.00b | 3.11s | +604.00ms ⚠️ |
| dist/component.37867f50.js | 22.51kb | +0.00b | 6.46s | -1.56s 🚀 |
| dist/EmojiPickerComponent.898d8be6.js | 18.23kb | +0.00b | 10.43s | -1.62s 🚀 |
| dist/js.48be70a4.js | 16.40kb | +0.00b | 6.46s | +1.00s ⚠️ |
| dist/ui.cee430a3.js | 13.97kb | +0.00b | 13.29s | -1.45m 🚀 |
| dist/smartMediaEditor.7ee0079c.js | 13.95kb | +0.00b | 1.71m | +43.64s ⚠️ |
| dist/workerHasher.65b8a1b7.js | 11.78kb | +0.00b | 55.55s | +41.88s ⚠️ |
| dist/component.11476a92.js | 6.05kb | +0.00b | 6.46s | +412.00ms ⚠️ |
| dist/card.4ac5c1b2.js | 5.61kb | +0.00b | 14.61s | -1.45s 🚀 |
| dist/index.393afe59.css | 3.46kb | +0.00b | 1.02m | +57.22s ⚠️ |
| dist/png-chunks-extract.8d15a914.js | 3.41kb | +0.00b | 6.46s | +996.00ms ⚠️ |
| dist/dropzone.2f1a1e17.js | 3.23kb | +0.00b | 53.02s | -2.81s 🚀 |
| dist/16.d1b8e6f7.js | 2.38kb | +0.00b | 3.09s | -2.36s 🚀 |
| dist/card.cadccd96.js | 2.03kb | +0.00b | 28.24s | -3.76s 🚀 |
| dist/ResourcedEmojiComponent.d9a49606.js | 2.00kb | +0.00b | 5.50s | -1.63s 🚀 |
| dist/16.f40c1329.js | 1.73kb | +0.00b | 6.80s | -1.59s 🚀 |
| dist/16.86a8c7d4.js | 1.66kb | +0.00b | 7.60s | +1.53s ⚠️ |
| dist/feedback.8b7c7928.js | 1.64kb | +0.00b | 7.50s | -2.73s 🚀 |
| dist/workerHasher.9bb425b1.js | 1.54kb | +0.00b | 55.54s | +41.87s ⚠️ |
| dist/16.cff410ff.js | 1.38kb | +0.00b | 6.65s | +345.00ms ⚠️ |
| dist/16.c1ed07f8.js | 1.33kb | +0.00b | 7.60s | +1.55s ⚠️ |
| dist/16.6603f2f3.js | 1.33kb | +0.00b | 6.65s | +345.00ms ⚠️ |
| dist/16.a4b392c2.js | 1.32kb | +0.00b | 3.09s | +724.00ms ⚠️ |
| dist/16.249c644a.js | 1.31kb | +0.00b | 6.95s | +395.00ms ⚠️ |
| dist/heading6.4c4017db.js | 1.30kb | +0.00b | 7.50s | -2.10s 🚀 |
| dist/16.e60b4d77.js | 1.21kb | +0.00b | 3.09s | +724.00ms ⚠️ |
| dist/heading5.dde15cec.js | 1.17kb | +0.00b | 7.50s | -2.10s 🚀 |
| dist/expand.6889aead.js | 1.16kb | +0.00b | 7.50s | -2.73s 🚀 |
| dist/media-card-analytics-error-boundary.6a25b970.js | 1.15kb | +0.00b | 53.02s | +21.02s ⚠️ |
| dist/16.2096b9f3.js | 1.14kb | +0.00b | 3.09s | +724.00ms ⚠️ |
| dist/16.f040d62d.js | 1.14kb | +0.00b | 6.65s | +345.00ms ⚠️ |
| dist/16.ecf8a702.js | 1.13kb | +0.00b | 7.67s | +1.61s ⚠️ |
| dist/16.1b342686.js | 1.13kb | +0.00b | 6.65s | +345.00ms ⚠️ |
| dist/16.2e7f52c6.js | 1.13kb | +0.00b | 6.65s | +343.00ms ⚠️ |
| dist/16.da43aa07.js | 1.13kb | +0.00b | 7.60s | +1.55s ⚠️ |
| dist/16.dd07505e.js | 1.10kb | +0.00b | 5.50s | +3.13s ⚠️ |
| dist/component.40194190.js | 1.07kb | +0.00b | 5.50s | -1.63s 🚀 |
| dist/media-card-analytics-error-boundary.964206cc.js | 833.00b | +0.00b | 13.29s | -1.50s 🚀 |
| dist/media-viewer-analytics-error-boundary.bbe40a02.js | 757.00b | +0.00b | 16.09s | -15.80s 🚀 |
| dist/EmojiPickerComponent.a968dafe.js | 535.00b | +0.00b | 10.43s | -45.76s 🚀 |
| dist/index.html | 119.00b | +0.00b | 841.00ms | -5.23s 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.4ca65e68.js | 2.73mb | +0.00b | 503.00ms | +62.00ms ⚠️ |
| dist/pdfRenderer.cd742f5e.js | 1.11mb | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/editorView.ae59b49f.js | 585.98kb | +0.00b | 499.00ms | +74.00ms ⚠️ |
| dist/popup.4a66e15e.js | 170.62kb | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/EmojiPickerComponent.bab709db.js | 97.89kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/Toolbar.33a83ae3.js | 92.34kb | +0.00b | 499.00ms | +74.00ms ⚠️ |
| dist/media-viewer.1d9322e7.js | 69.09kb | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/card.ce67dc35.js | 55.09kb | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/card.1007a95f.js | 52.86kb | +0.00b | 499.00ms | +73.00ms ⚠️ |
| dist/Modal.e3bd77d0.js | 41.40kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/component.59edff1f.js | 30.78kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/component.37867f50.js | 22.51kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/DatePicker.cbe44a9b.js | 21.17kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/EmojiPickerComponent.898d8be6.js | 18.23kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/js.48be70a4.js | 16.40kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/dropzone.c8c03b6a.js | 14.36kb | +0.00b | 499.00ms | +73.00ms ⚠️ |
| dist/ui.cee430a3.js | 13.97kb | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/smartMediaEditor.7ee0079c.js | 13.95kb | +0.00b | 499.00ms | +74.00ms ⚠️ |
| dist/workerHasher.65b8a1b7.js | 11.78kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/EmojiTypeAheadComponent.9af6e5fc.js | 8.67kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/component.11476a92.js | 6.05kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/card.4ac5c1b2.js | 5.61kb | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/media-viewer.bc661bce.js | 3.92kb | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/EmojiPickerComponent.624bf3f4.js | 3.58kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/index.393afe59.css | 3.46kb | +0.00b | 499.00ms | +74.00ms ⚠️ |
| dist/png-chunks-extract.8d15a914.js | 3.41kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/dropzone.2f1a1e17.js | 3.23kb | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/Modal.2835744c.js | 3.08kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/clipboard.0244488f.js | 2.75kb | +0.00b | 499.00ms | +73.00ms ⚠️ |
| dist/16.d1b8e6f7.js | 2.38kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/EmojiUploadComponent.88a444df.js | 2.18kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/card.cadccd96.js | 2.03kb | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/ResourcedEmojiComponent.d9a49606.js | 2.00kb | +0.00b | 503.00ms | +62.00ms ⚠️ |
| dist/date.d49e64f8.js | 1.73kb | +0.00b | 500.00ms | +60.00ms ⚠️ |
| dist/16.f40c1329.js | 1.73kb | +0.00b | 502.00ms | +63.00ms ⚠️ |
| dist/images.80270b1f.js | 1.67kb | +0.00b | 501.00ms | +61.00ms ⚠️ |
| dist/16.86a8c7d4.js | 1.66kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/feedback.8b7c7928.js | 1.64kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/browser.16c438d4.js | 1.61kb | +0.00b | 499.00ms | +73.00ms ⚠️ |
| dist/workerHasher.9bb425b1.js | 1.54kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/workerHasher.f66f00c7.js | 1.54kb | +0.00b | 499.00ms | +73.00ms ⚠️ |
| dist/list-number.4632b459.js | 1.46kb | +0.00b | 500.00ms | +60.00ms ⚠️ |
| dist/status.cccf04be.js | 1.46kb | +0.00b | 501.00ms | +62.00ms ⚠️ |
| dist/16.cff410ff.js | 1.38kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/code.af9ca2b3.js | 1.38kb | +0.00b | 501.00ms | +61.00ms ⚠️ |
| dist/16.c1ed07f8.js | 1.33kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/16.6603f2f3.js | 1.33kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/16.a4b392c2.js | 1.32kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/16.249c644a.js | 1.31kb | +0.00b | 501.00ms | +61.00ms ⚠️ |
| dist/link.d12cc130.js | 1.30kb | +0.00b | 500.00ms | +60.00ms ⚠️ |
| dist/heading6.4c4017db.js | 1.30kb | +0.00b | 501.00ms | +62.00ms ⚠️ |
| dist/heading3.93606b9c.js | 1.29kb | +0.00b | 501.00ms | +62.00ms ⚠️ |
| dist/16.ab5c0088.js | 1.27kb | +0.00b | 501.00ms | +61.00ms ⚠️ |
| dist/emoji.e603e70e.js | 1.23kb | +0.00b | 501.00ms | +61.00ms ⚠️ |
| dist/16.afee1494.js | 1.23kb | +0.00b | 502.00ms | +63.00ms ⚠️ |
| dist/16.e60b4d77.js | 1.21kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/16.7c7a40c9.js | 1.19kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/16.06d5a67b.js | 1.18kb | +0.00b | 501.00ms | +61.00ms ⚠️ |
| dist/heading5.dde15cec.js | 1.17kb | +0.00b | 501.00ms | +62.00ms ⚠️ |
| dist/expand.6889aead.js | 1.16kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.6a25b970.js | 1.15kb | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/16.2096b9f3.js | 1.14kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/16.f040d62d.js | 1.14kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/16.ecf8a702.js | 1.13kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/16.1b342686.js | 1.13kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/16.2e7f52c6.js | 1.13kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/16.da43aa07.js | 1.13kb | +0.00b | 502.00ms | +62.00ms ⚠️ |
| dist/heading2.e219309a.js | 1.11kb | +0.00b | 501.00ms | +62.00ms ⚠️ |
| dist/16.dd07505e.js | 1.10kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/component.40194190.js | 1.07kb | +0.00b | 502.00ms | +61.00ms ⚠️ |
| dist/mention.b1828447.js | 1.07kb | +0.00b | 500.00ms | +60.00ms ⚠️ |
| dist/heading4.5316f6f2.js | 1.06kb | +0.00b | 501.00ms | +62.00ms ⚠️ |
| dist/16.317b9164.js | 1.05kb | +0.00b | 501.00ms | +61.00ms ⚠️ |
| dist/layout.ac6734f9.js | 1.05kb | +0.00b | 501.00ms | +61.00ms ⚠️ |
| dist/divider.d2be55a1.js | 1.03kb | +0.00b | 500.00ms | +60.00ms ⚠️ |
| dist/quote.f4e09143.js | 1.02kb | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/action.cb287eb0.js | 1.00kb | +0.00b | 501.00ms | +61.00ms ⚠️ |
| dist/decision.91f3af14.js | 1010.00b | +0.00b | 500.00ms | +60.00ms ⚠️ |
| dist/panel-warning.68b04096.js | 1008.00b | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/list.057cb664.js | 978.00b | +0.00b | 500.00ms | +60.00ms ⚠️ |
| dist/heading1.fa66b375.js | 976.00b | +0.00b | 501.00ms | +62.00ms ⚠️ |
| dist/panel-error.c4b05f6f.js | 906.00b | +0.00b | 500.00ms | +60.00ms ⚠️ |
| dist/panel.9dbc048e.js | 897.00b | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/table.b35545be.js | 889.00b | +0.00b | 501.00ms | +62.00ms ⚠️ |
| dist/panel-success.7eb0809a.js | 845.00b | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/panel-note.ec8386b7.js | 841.00b | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/media-picker-analytics-error-boundary.553055f6.js | 837.00b | +0.00b | 499.00ms | +73.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.964206cc.js | 833.00b | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.57503222.js | 833.00b | +0.00b | 499.00ms | +73.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.bbe40a02.js | 757.00b | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/EmojiPickerComponent.a968dafe.js | 535.00b | +0.00b | 500.00ms | +61.00ms ⚠️ |
| dist/simpleHasher.f8e69464.js | 528.00b | +0.00b | 500.00ms | +74.00ms ⚠️ |
| dist/simpleHasher.32d9352e.js | 528.00b | +0.00b | 499.00ms | +73.00ms ⚠️ |
| dist/index.html | 119.00b | +0.00b | 504.00ms | +46.00ms ⚠️ |

</p></details>

<details><summary>Three.js x4 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/271387638582215180)**

File: packages/core/parcel/src/cli.js (hunk 1 above)
Comment (DeMoorJasper) [reply packages/core/parcel/src/cli.js:273]: Are you sure? Don't we want to allow this in the api?

Comment (DeMoorJasper) [reply packages/core/parcel/src/cli.js:273]: Just changed the default to false when using the API, this way anyone can still enable it when using the API but only explicitly

Comment (mischnic) [reply packages/core/parcel/src/cli.js:273]: I was thinking of `autoinstall: initialOptions.autoinstall ?? mode !== 'production',` but this is fine as well

**Pull request merged**